packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -380,12 +445,18 @@ const findTokenAccountOwner = (
 ): SubscriptionAccountInfo | undefined =>
     accounts.find(account =>
         account.tokens?.find(token =>
-            token.accounts?.find(
-                tokenAccount => tokenAccount.publicKey.toString() === accountDescriptor,
-            ),
+            token.accounts?.find(tokenAccount => tokenAccount.publicKey === accountDescriptor),
         ),
     );
 
+let NEXT_ACCOUNT_SUBSCRIPTION_ID = 0;
+const ACCOUNT_SUBSCRIPTION_ABORT_CONTROLLERS = new Map<number, AbortController>();
+function abortSubscription(id: number) {
+    const abortController = ACCOUNT_SUBSCRIPTION_ABORT_CONTROLLERS.get(id);
+    ACCOUNT_SUBSCRIPTION_ABORT_CONTROLLERS.delete(id);
+    abortController?.abort();
+}
```

Comment:
I would have preferred to put an `unsubscribe()` function on the `SubscriptionAccountInfo` type itself so that I didn't have to establish this correspondence between subscription ‘ids’ and abortcontrollers, but that type is in  common so I didn't touch it.